adapter/validator/src/test/java/org/mobilitydata/gtfsvalidator/validator/GtfsTypeValidatorTest.java

```diff
@@ -167,870 +218,927 @@ void outOfRangeInt_decGenerateNotice() throws IOException {
-                TEST_ID,
-                Map.of("int_hex_with_regex", "ABCDEF0"),
-                new RawFileInfo.RawFileInfoBuilder().filename(TEST_FILE_TST).build()
-        ));
+        verify(mockColorValidator, times(1)).isValid(ArgumentMatchers.eq("AZ-FTJ"));
+    }
 
-        assertEquals(1, result.size());
+    @Test
+    void timeInCorrectFormatDoNotGenerateNotice() {
 
-        notice = new ArrayList<>(result).get(0);
-        assertThat(notice, instanceOf(InvalidColorNotice.class));
-        assertEquals("E014", notice.getId());
-        assertEquals("Invalid color", notice.getTitle());
-        assertEquals(TEST_FILE_TST, notice.getFilename());
-        assertEquals("Invalid color:ABCDEF0 in field:int_hex_with_regex for entity with id:test_id",
-                notice.getDescription());
+        RegexValidator mockTimeValidator = mock(RegexValidator.class);
+        when(mockTimeValidator.isValid(ArgumentMatchers.any())).thenReturn(true);
+
+        GtfsSpecificationProto.CsvSpecProto mockFileSpec = mock(GtfsSpecificationProto.CsvSpecProto.class);
+        GtfsSpecificationProto.ColumnSpecProto mockColumnSpec = mock(GtfsSpecificationProto.ColumnSpecProto.class);
+        when(mockColumnSpec.getName()).thenReturn("time_with_regex");
+
+        GtfsSpecificationProto.ColumnInputType mockInputType = mock(GtfsSpecificationProto.ColumnInputType.class);
+        when(mockInputType.getType()).thenReturn(GtfsSpecificationProto.ColumnInputType.InputType.TIME);
+
+        when(mockColumnSpec.getType()).thenReturn(mockInputType);
+        when(mockFileSpec.getColumnList()).thenReturn(List.of(mockColumnSpec));
 
-        // invalid characters
-        result = underTest.validate(new ParsedEntity(
+        GtfsTypeValidator underTest = new GtfsTypeValidator(mockFileSpec,
+                mock(FloatValidator.class),
+                mock(IntegerValidator.class),
+                mock(UrlValidator.class),
+                mock(RegexValidator.class),
+                mockTimeValidator,
+                Collections.emptySet()
+        );
+
+        Collection<Notice> result = underTest.validate(new ParsedEntity(
                 TEST_ID,
-                Map.of("int_hex_with_regex", "AZ-FTJ"),
+                Map.of("time_with_regex", "01:02:03"),
                 new RawFileInfo.RawFileInfoBuilder().filename(TEST_FILE_TST).build()
         ));
 
-        assertEquals(1, result.size());
+        assertEquals(0, result.size());
 
-        notice = new ArrayList<>(result).get(0);
-        assertThat(notice, instanceOf(InvalidColorNotice.class));
-        assertEquals("E014", notice.getId());
-        assertEquals("Invalid color", notice.getTitle());
-        assertEquals(TEST_FILE_TST, notice.getFilename());
-        assertEquals("Invalid color:AZ-FTJ in field:int_hex_with_regex for entity with id:test_id",
-                notice.getDescription());
+        verify(mockTimeValidator, times(1)).isValid(ArgumentMatchers.eq("01:02:03"));
```

Comment:
Time is tricky in GTFS - the [spec says](https://gtfs.org/reference/static/#field-types):

>Time - Time in the HH:MM:SS format (H:MM:SS is also accepted). The time is measured from "noon minus 12h" of the service day (effectively midnight except for days on which daylight savings time changes occur). For times occurring after midnight, enter the time as a value greater than 24:00:00 in HH:MM:SS local time for the day on which the trip schedule begins.
Example: 14:30:00 for 2:30PM or 25:35:00 for 1:35AM on the next day.

There need to be a few more assertions here to make sure we're covering all the valid times.

I suggest verifying that the following are correctly parsed as legit times:
* "22:15:09"
* "25:05:55"
* "5:15:35"

Reply:
Ok, in those unit tests the isValid return value is mocked and the parameter check is just here to verify it is passed unchanged. That way we don't test the validator library code at the same time. Do you think we shall have integration tests as well?